manifests/init.pp

```diff
@@ -252,14 +252,18 @@
   $cluster_node_timeout                                          = $redis::params::cluster_node_timeout,
   Integer[0] $cluster_slave_validity_factor                      = $redis::params::cluster_slave_validity_factor,
   Boolean $cluster_require_full_coverage                         = $redis::params::cluster_require_full_coverage,
-  Integer[0] $cluster_migration_barrier                          = $redis::params::cluster_migration_barrier
+  Integer[0] $cluster_migration_barrier                          = $redis::params::cluster_migration_barrier,
+  ## Start hiera lookup
+  $instances                                                     = {},
 ) inherits redis::params {
 
   contain redis::preinstall
   contain redis::install
   contain redis::config
   contain redis::service
 
+  create_resources('::redis::instance', $instances)
```

Comment:
Instead of `create_resources`, could you use iteration instead? (Also drop the leading `::`).
See https://github.com/voxpupuli/puppet-firewalld/blob/master/manifests/init.pp#L103 for an example.

Reply:
Again, thank you. Done.